src/main/java/org/opentripplanner/routing/algorithm/profile/OptimizationProfileFactory.java

```diff
@@ -0,0 +1,32 @@
+package org.opentripplanner.routing.algorithm.profile;
+
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class OptimizationProfileFactory {
```

Comment:
wydaje mi się, że tworzenie nowej instancji tej klasy za każdym razem jak potrzebujemy `OptimizationProfile` to trochę przesada ;). może niech po prostu metoda `getOptimizationProfile` będzie statyczna?